tests/ReflectorFactory.php

```diff
@@ -64,7 +65,7 @@ public static function create(string $cacheDir): QueryReflector
                 $schemaHasher = new SchemaHasherMysql($pdo);
             } elseif ('pdo-pgsql' === $reflector) {
```

Comment:
Yeah I agree, as mentioned earlier I'd refactor the current PDO reflector to use the new base reflector - as to what you mentioned before - I'm currently not planning on adding more stuff to the base class, as the common logic for the both reflector should be there now - so it's just a matter of dropping it from the current PDO reflector mostly.

But maybe I'll do it separately once we finish the PG support?

As to the current test-suite run - it looks similar to what we can see in the github action PR, but as this MR helps to resolve the proper types and flags, it's closer to correct results e.g.:

```
Expected :'Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<0, 4294967295>, 1: int<0, 4294967295>, gesperrt: int<-128, 127>, 2: int<-128, 127>, freigabe1u1: int<-128, 127>, 3: int<-128, 127>}>'
Actual   :'Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<1, 2147483647>, 1: int<1, 2147483647>, gesperrt: int<-2147483648, 2147483647>, 2: int<-2147483648, 2147483647>, freigabe1u1: int<-32768, 32767>, 3: int<-32768, 32767>}>'
```

instead of:
```
Expected :'Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<0, 4294967295>, 1: int<0, 4294967295>, gesperrt: int<-128, 127>, 2: int<-128, 127>, freigabe1u1: int<-128, 127>, 3: int<-128, 127>}>'
Actual   :'Doctrine\DBAL\Result'
```
what you might be getting in the github action PR

I'm wondering if we should prepare a different test-suite for the pgsql reflector - with pgsql ranges etc. or just do something like:

```
$isPgsql = $conn->getDatabasePlatform() instanceof PostgreSQLPlatform::class;
$result = $conn->query('SELECT email, adaid, gesperrt, freigabe1u1 FROM ada');

if ($isPgsql) {
   assertType('Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<1, 2147483647>, 1: int<1, 2147483647>, gesperrt: int<-2147483648, 2147483647>, 2: int<-2147483648, 2147483647>, freigabe1u1: int<-32768, 32767>, 3: int<-32768, 32767>}>', $result)
} else {
    assertType('Doctrine\DBAL\Result<array{email: string, 0: string, adaid: int<0, 4294967295>, 1: int<0, 4294967295>, gesperrt: int<-128, 127>, 2: int<-128, 127>, freigabe1u1: int<-128, 127>, 3: int<-128, 127>}>', $result);
}
```

Reply:
maybe i don't understand the test question completely, i see you already merged/rebased this, so test results are here..

> or does it work locally with inferred types already? it looks like in the github action the inferred databased types do not work.

Locally it's the same

Reply:
Ohh I looked into the wrong action job (the replay job), in the regular postgres jobs things look good.

> I'm wondering if we should prepare a different test-suite for the pgsql reflector - with pgsql ranges etc. or just do something like:


i think we can change tests, which don‘t test the actual types but just test whether types are availabe, with types which work consistently across pgsql/mysql.


A single test should be used to test the actual column types, like https://github.com/staabm/phpstan-dba/pull/130/files

Reply:
Acutally I am not 100% sure. But we should prevent assertion of different types as good as possible, when not in a test which actually tests the column types

Reply:
I have adjusted the existing test-suite in a few PRs on the master branch.
this made the test-suite pass for pgsql, for most but not all tests.

the remaining stuff not passing right now seems to be legit problems which need to be investigated

Reply:
Hey, I'm sorry for leaving this without my attention for so long. I'm kind of getting different results in the test-suite than what was left here by you. Do you know what is it you did regarding the test-case adjustments? I must have screwed up something 😞

Reply:
Perhaps I'm just confused, probably all is here..

Reply:
I think you need to work thru the tests one by one of this job: https://github.com/staabm/phpstan-dba/runs/5709972883?check_suite_focus=true

it seems some types are not properly inferred/mapped from the db.
also it seems something is wrong regarding NULLability. NULL seems to show up at too many types

Reply:
@staabm I think I might be getting close to the finish line, but there's still a couple of unresolved things:

1.) the `typemix` database table in tests for mysql includes basically all mysql types - it's not possible to have an exact copy for psql, e.g. with no unsigned ints and other type differences. I'm not sure if we haven't already reduced the amount of columns there for postgres, I've included a copy of the mysql database as close as it can get.

2.) phpstan version - static analysis is now failing for tests, but with the latest release all SA checks are ok (^1.5), locally I get to 1.4.5 after installing dependencies, which is below the mentioned version required in tests (1.4.6+)

3.) For the rest I've temporarily included some comments
- one null result I still need to inspect further, might be possible to fix it somehow
- one failing query with an operator not available for psql
- one pdo statement result format issue
- integer ranges

Reply:
Regarding 1) feel free to make this test skip on pgsql. Its all about mysql specific stuff/types

Regarding 2) feel free to mark such cases as beeing skipped for pfsql for now. We don't need 100% feature parity for pgsql and mysql. 

We should keep the mysql test coverage the same as on master-branch

Regarding 3) thx I will have a look